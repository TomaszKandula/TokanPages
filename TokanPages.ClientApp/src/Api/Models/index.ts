export type { AddArticleDto } from "./Articles/addArticleDto";
export type { UpdateArticleContentDto } from "./Articles/updateArticleContentDto";
export type { UpdateArticleCountDto } from "./Articles/updateArticleCountDto";
export type { UpdateArticleLikesDto } from "./Articles/updateArticleLikesDto";
export type { UpdateArticleVisibilityDto } from "./Articles/updateArticleVisibilityDto";
export type { RemoveArticleDto } from "./Articles/removeArticleDto";

export type { SendMessageDto } from "./Mailer/sendMessageDto";
export type { SendNewsletterDto } from "./Mailer/sendNewsletterDto";
export type { SubscriberInfoDto } from "./Mailer/subscriberInfoDto";
export type { VerifyEmailAddressDto } from "./Mailer/verifyEmailAddressDto";

export type { AddSubscriberDto } from "./Subscribers/addSubscriberDto";
export type { UpdateSubscriberDto } from "./Subscribers/updateSubscriberDto";
export type { RemoveSubscriberDto } from "./Subscribers/removeSubscriberDto";

export type { UserDataDto } from "./Users/userDataDto";
export type { ActivateUserDto } from "./Users/activateUserDto";
export type { AddUserDto } from "./Users/addUserDto";
export type { UpdateUserDto } from "./Users/updateUserDto";
export type { RemoveUserDto } from "./Users/removeUserDto";
export type { AuthenticateUserDto } from "./Users/authenticateUserDto";
export type { AuthenticateUserResultDto } from "./Users/authenticateUserResultDto";
export type { ReAuthenticateUserDto } from "./Users/reAuthenticateUserDto";
export type { ReAuthenticateUserResultDto } from "./Users/reAuthenticateUserResultDto";
export type { RevokeUserRefreshTokenDto } from "./Users/revokeUserRefreshTokenDto";
export type { ResetUserPasswordDto } from "./Users/resetUserPasswordDto";
export type { UpdateUserPasswordDto } from "./Users/updateUserPasswordDto";
export type { UserPermissionDto } from "./Users/userPermissionDto";
export type { UserRoleDto } from "./Users/userRoleDto";
export type { UploadUserMediaDto } from "./Users/uploadUserMediaDto";
export type { UploadUserMediaResultDto } from "./Users/uploadUserMediaResultDto";

export type { ErrorDto } from "./Error/errorDto";
export type { ValidationErrorsDto } from "./Error/validationErrorsDto";

export type { DocumentContentDto } from "./Components/documentContentDto";

export type { NavigationContentDto } from "./Components/navigationContentDto";
export type { HeaderContentDto } from "./Components/headerContentDto";
export type { FooterContentDto } from "./Components/footerContentDto";
export type { ClientsContentDto } from "./Components/clientsContentDto";
export type { ArticleFeaturesContentDto } from "./Components/articleFeaturesContentDto";
export type { ContactFormContentDto } from "./Components/contactFormContentDto";
export type { CookiesPromptContentDto } from "./Components/cookiesPromptContentDto";
export type { FeaturedContentDto } from "./Components/featuredContentDto";
export type { FeaturesContentDto } from "./Components/featuresContentDto";
export type { NewsletterContentDto } from "./Components/newsletterContentDto";
export type { ResetPasswordContentDto } from "./Components/resetPasswordContentDto";
export type { UpdatePasswordContentDto } from "./Components/updatePasswordContentDto";
export type { UserSigninContentDto } from "./Components/userSigninContentDto";
export type { UserSignoutContentDto } from "./Components/userSignoutContentDto";
export type { UserSignupContentDto } from "./Components/userSignupContentDto";
export type { TestimonialsContentDto } from "./Components/testimonialsContentDto";
export type { UnsubscribeContentDto } from "./Components/unsubscribeContentDto";
export type { ActivateAccountContentDto } from "./Components/activateAccountContentDto";
export type { UpdateSubscriberContentDto } from "./Components/updateSubscriberContentDto";
export type { WrongPagePromptContentDto } from "./Components/wrongPagePromptContentDto";
export type { 
    AccountContentDto, 
    SectionAccessDenied,
    SectionAccountInformation,
    SectionAccountPassword,
    SectionAccountRemoval
} from "./Components/accountContentDto";

export type { IconDto } from "./Components/Common/iconDto";
export type { LinkDto } from "./Components/Common/linkDto";
export type { RowItemDto } from "./Components/Common/rowItemDto";
export type { TextItemDto } from "./Components/Common/textItemDto";
export type { ContentDto } from "./Components/Common/contentDto";

export type { GetContentManifestDto } from "./Content/getContentManifestDto";
export type { LanguageItemDto } from "./Content/Items/languageItemDto";
export type { JWT } from "./Jwt";
