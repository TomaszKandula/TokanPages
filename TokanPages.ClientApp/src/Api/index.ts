export {
    IsSuccessStatusCode,
    GetBaseHeaders,
    GetProcessedBody,
    GetProcessedHeaders,
    GetProcessedResponse,
} from "./Utilities";

export type { ExecuteApiActionResultProps, ExecuteStoreActionProps, ExecuteApiActionProps } from "./Utilities/Types";

export {
    API_BASE_URI,
    GET_ARTICLES,
    GET_ARTICLE,
    GET_ARTICLE_BY_TITLE,
    ADD_ARTICLE,
    UPDATE_ARTICLE_CONTENT,
    UPDATE_ARTICLE_COUNT,
    UPDATE_ARTICLE_LIKES,
    UPDATE_ARTICLE_VISIBILITY,
    RETRIEVE_ARTICLE_INFO,
    REMOVE_ARTICLE,
    LOG_MESSAGE,
    GET_USERS,
    GET_USER,
    GET_USER_IMAGE,
    ADD_USER,
    ACTIVATE_USER,
    UPDATE_USER,
    REMOVE_USER,
    AUTHENTICATE,
    REAUTHENTICATE,
    VERIFY_USER_EMAIL,
    REVOKE_REFRESH_TOKEN,
    REVOKE_USER_TOKEN,
    RESET_USER_PASSWORD,
    UPDATE_USER_PASSWORD,
    UPLOAD_USER_IMAGE,
    GET_NEWSLETTERS,
    GET_NEWSLETTER,
    ADD_NEWSLETTER,
    UPDATE_NEWSLETTER,
    REMOVE_NEWSLETTER,
    SEND_MESSAGE,
    SEND_NEWSLETTER,
    NOTIFY_WEB_URL,
    NOTIFICATION_STATUS,
    GET_CONTENT_MANIFEST,
    REQUEST_PAGE_DATA,
    GET_ARTICLE_MAIN_IMAGE_URL,
    GET_IMAGES_URL,
    GET_DOCUMENTS_URL,
    ARTICLE_PATH,
    GET_USER_VIDEO,
    GET_VIDEO_ASSET,
    GET_USER_NOTES,
    GET_USER_NOTE,
    ADD_USER_NOTE,
    UPDATE_USER_NOTE,
    REMOVE_USER_NOTE,
} from "./Paths";
