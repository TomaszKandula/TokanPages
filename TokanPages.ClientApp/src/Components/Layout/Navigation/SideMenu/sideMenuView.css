.sidemenu-drawer-container {
    width: 300px;
}

.sidemenu-drawer-hero {
    height: 200px;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: #ebecfb;
}

.sidemenu-logo {
    height: 120px;
    margin-left: auto;
    margin-right: auto;
    align-self: center;
}
