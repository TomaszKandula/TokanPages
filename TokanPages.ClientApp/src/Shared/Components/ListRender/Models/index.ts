export type { Fields } from "./Fields";
export type { Item } from "./Item";
export type { Subitem } from "./Subitem";
