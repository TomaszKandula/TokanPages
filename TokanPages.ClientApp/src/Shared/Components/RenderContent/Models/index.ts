export type { IArticleItem } from "./ArticleItemModel";
export type { IAuthor } from "./AuthorModel";
export type { ITextObject, ITextItem, IRowItem } from "./TextModel";
