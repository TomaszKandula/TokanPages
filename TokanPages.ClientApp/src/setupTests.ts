// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
