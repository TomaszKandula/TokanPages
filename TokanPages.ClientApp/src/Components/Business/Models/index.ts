import { ServiceItemDto, TechItemsDto } from "../../../Api/Models";
import { ReactChangeEvent, ReactMouseEvent } from "../../../Shared/types";

export interface MessageFormProps {
    company: string;
    firstName: string;
    lastName: string;
    email: string;
    phone: string;
    description: string;
    techStack: string[];
    services: string[];
}

export interface TechStackListProps {
    isDisabled: boolean;
    list: TechItemsDto[];
    handler: (event: ReactChangeEvent, isChecked: boolean) => void;
}

export interface ServiceItemCardProps {
    isDisabled: boolean;
    key: React.Key | null | undefined;
    value: ServiceItemDto;
    handler: (event: ReactMouseEvent) => void;
    services: string[];
}

export interface BusinessFormProps {
    hasCaption?: boolean;
    hasIcon?: boolean;
    hasShadow?: boolean;
    className?: string;
    background?: string;
}
