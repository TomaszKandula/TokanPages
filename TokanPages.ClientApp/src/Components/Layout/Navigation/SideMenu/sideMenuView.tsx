import * as React from "react";
import { Drawer } from "@material-ui/core";
import { RenderImage, RenderSideMenu } from "../../../../Shared/Components";
import { Item } from "../../../../Shared/Components/RenderMenu/Models";
import { GET_ICONS_URL } from "../../../../Api/Request";

interface Properties {
    drawerState: { open: boolean };
    closeHandler: any;
    isAnonymous: boolean;
    languageId: string;
    menu: { image: string; items: Item[] };
}

export const SideMenuView = (props: Properties): React.ReactElement => {
    return (
        <Drawer anchor="left" open={props.drawerState.open} onClose={props.closeHandler}>
            <div className="sidemenu-drawer-container">
                <div className="sidemenu-drawer-hero">
                    <RenderImage basePath={GET_ICONS_URL} imageSource={props?.menu?.image} className="sidemenu-logo" />
                </div>
                <RenderSideMenu
                    isAnonymous={props.isAnonymous}
                    languageId={props.languageId}
                    items={props.menu?.items}
                />
            </div>
        </Drawer>
    );
};
