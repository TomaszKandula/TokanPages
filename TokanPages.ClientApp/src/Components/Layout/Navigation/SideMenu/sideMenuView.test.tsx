import "../../../../setupTests";
import React from "react";
import { render } from "@testing-library/react";
import { ItemDto } from "../../../../Api/Models";
import { SideMenuView } from "./sideMenuView";

describe("test component: menuView", () => {
    it("should render correctly '<MenuView />' when content is loaded.", () => {
        const item: ItemDto = {
            id: "8c678e9e-699e-41c7-a93a-587c6a4f41e7",
            type: "item",
            value: "Sign in",
            link: "/en/signin",
            icon: "VpnKey",
            enabled: true,
        };

        const html = render(
            <SideMenuView
                drawerState={{ open: false }}
                closeHandler={jest.fn()}
                isAnonymous={true}
                languageId="en"
                menu={{ image: "", items: [item] }}
            />
        );

        expect(html).toMatchSnapshot();
    });
});
