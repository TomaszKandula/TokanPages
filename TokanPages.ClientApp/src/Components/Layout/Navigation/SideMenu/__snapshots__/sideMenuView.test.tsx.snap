// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component: menuView should render correctly '<MenuView />' when content is loaded. 1`] = `
<WithStyles(ForwardRef(Drawer))
  anchor="left"
  onClose={[MockFunction]}
  open={false}
>
  <div
    className="makeStyles-drawer_container-2"
  >
    <Styled(MuiBox)
      className="makeStyles-drawer_hero-3"
    />
    <RenderList
      isAnonymous={true}
      items={
        Array [
          Object {
            "enabled": true,
            "icon": "VpnKey",
            "id": "8c678e9e-699e-41c7-a93a-587c6a4f41e7",
            "link": "/signin",
            "type": "item",
            "value": "Sign in",
          },
        ]
      }
    />
  </div>
</WithStyles(ForwardRef(Drawer))>
`;
